hasktorch/src/Torch/Typed/Native.hs

```diff
@@ -2943,9 +2943,72 @@ qZeroPoint input = unsafePerformIO $ cast1 ATen.q_zero_point_t input
 -- combinations :: Tensor device dtype shape -> Int -> Bool -> Tensor device dtype shape
 -- combinations _input _r _with_replacement = unsafePerformIO $ (cast3 ATen.combinations_tlb) _input _r _with_replacement
 
--- | lstmCell
--- >>> dtype &&& shape $ fst $ lstmCell (ones :: CPUTensor 'D.Float '[12,2]) (ones :: CPUTensor 'D.Float '[12,3]) (ones :: CPUTensor 'D.Float '[12]) (ones :: CPUTensor 'D.Float '[12]) ((ones :: CPUTensor 'D.Float '[2,3]), (ones :: CPUTensor 'D.Float '[2,3])) (ones :: CPUTensor 'D.Float '[2,2])
--- (Float,[2,3])
+-- | lstm
+-- Parameters for this ATen function are non-trivially provided.  See the
+-- `Typed.NN.LSTM` module for doctests.
+--
+lstm
+  :: forall
+       numLayers
+       numDirections
+       dtype
+       seqLen
+       batchSize
+       inputDim
+       hiddenSize
+       device
+   . (KnownNat numLayers, KnownNat numDirections, KnownNat hiddenSize)
+  => Tensor device dtype '[seqLen, batchSize, inputDim]
+  -> ( Tensor
+         device
+         dtype
+         '[numLayers * numDirections, batchSize, hiddenSize]
+     , Tensor
+         device
+         dtype
+         '[numLayers * numDirections, batchSize, hiddenSize]
+     )
+  -> [D.Tensor]
```

Comment:
ideally, there would be no reference to `D.Tensor` in all signatures in `Torch.Typed.Native`.
if we used `HList`s here, we could make that happen. This could be done later, though.

Reply:
Right, that makes sense but probably better done when / if someone get's the itch to make the inductive `LSTM` bits `HList`s.

Reply:
Though the latter would also make `Generics` easier, and `Generics` make life in production much nicer.